lib/rubocop/cop/rails/reflection_class_name_string.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # This cop checks if the value of the option `class_name`, in
+      # the definition of a reflection is a string.
+      #
+      # @example
+      #   # bad
+      #   has_many :accounts, class_name: Account
+      #   has_many :accounts, class_name: Account.name
+      #
+      #   # good
+      #   has_many :accounts, class_name: 'Account'
+      class ReflectionClassNameString < Cop
```

Comment:
Like [`Rails/ActionFilter` cop](https://github.com/rubocop-hq/rubocop/blob/v0.63.1/lib/rubocop/cop/rails/action_filter.rb), I think the name `Rails/ReflectionClassName` is simple and clear.
The name inspired by original proposal name and https://github.com/rails/rails/pull/27551.

Reply:
You're right, `Rails/ReflectionClassName` is simple and clear enough.

Reply:
And thanks for those links. 😄